storm-core/src/jvm/org/apache/storm/command/AdminCommands.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.storm.cluster.ClusterUtils;
+import org.apache.storm.cluster.DaemonType;
+import org.apache.storm.cluster.IStormClusterState;
+import org.apache.storm.utils.Utils;
+import org.apache.storm.zookeeper.Zookeeper;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.storm.utils.ConfigUtils;
+
+import java.util.*;
+
+public class AdminCommands {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Deactivate.class);
+    private static ClientBlobStore clientBlobStore;
+    private static IStormClusterState stormClusterState;
+    private static CuratorFramework zk;
+    private static Map conf;
+
+    public static void main(String [] args) throws Exception {
+
+        if (args.length == 0) {
+            throw new IllegalArgumentException("Missing command.");
+        }
+        initialize();
+        String command = args[0];
+        String[] newArgs = Arrays.copyOfRange(args, 1, args.length);
+        switch (command) {
+            case "remove_corrupt_topologies":
+                removeCorruptTopologies();
+                break;
+            default:
+                throw new RuntimeException("" + command + " is not a supported admin command");
+        }
+
+    }
+
+    private static void initialize() {
+        Map conf = ConfigUtils.readStormConfig();
+        ClientBlobStore clientBlobStore = Utils.getClientBlobStore(conf);
+        List<String> servers = (List<String>) conf.get(Config.STORM_ZOOKEEPER_SERVERS);
+        Object port = conf.get(Config.STORM_ZOOKEEPER_PORT);
+        List<ACL> acls = null;
+        if (Utils.isZkAuthenticationConfiguredStormServer(conf)) {
+            acls = adminZkAcls();
+        }
+        try {
+            IStormClusterState stormClusterState = ClusterUtils.mkStormClusterState(conf, acls, new ClusterStateContext(DaemonType.UNKNOWN));
+        } catch (Exception e) {
+            LOG.error("admin can't create stormClusterState");
+            throw Utils.wrapInRuntime(e);
+        }
+        CuratorFramework zk = Zookeeper.mkClient(conf, servers, port, "", new DefaultWatcherCallBack(),conf);
+    }
+
+    // we might think of moving this method in Utils class
+    private static List<ACL> adminZkAcls() {
+        final List<ACL> acls = new ArrayList<>();
+        acls.add(ZooDefs.Ids.CREATOR_ALL_ACL.get(0));
+        acls.add(new ACL((ZooDefs.Perms.READ ^ ZooDefs.Perms.CREATE), ZooDefs.Ids.ANYONE_ID_UNSAFE));
+        return acls;
+    }
+
+    private static void removeCorruptTopologies( ) {
+        Iterator<String> corruptTopologies = listCorruptTopologies();
+        while(corruptTopologies.hasNext()) {
+            stormClusterState.removeStorm(corruptTopologies.next());
+        }
+    }
+
+    private static Iterator<String> listCorruptTopologies() {
+        Iterator<String> blobStoreTopologyIds = clientBlobStore.listKeys();
```

Comment:
This is very different from the code-ids function
```
(defn code-ids [blob-store]
  (let [to-id (reify KeyFilter
                (filter [this key] (ConfigUtils/getIdFromBlobKey key)))]
(set (.filterAndListKeys blob-store to-id))))
```

This lists all of the keys in the blob store.  code-ids Converts the key into a topology id.